aten/src/ATen/native/transformers/attention.cpp

```diff
@@ -692,30 +695,57 @@ std::tuple<Tensor, Tensor> _scaled_dot_product_attention(
 }
 
 std::tuple<Tensor, Tensor> _scaled_dot_product_attention_forward_math(
-        const Tensor& query_, const Tensor& key, const Tensor& value,
-        const c10::optional<Tensor>& attn_mask_, double dropout_p, bool need_attn_weights, bool is_causal){
-        return at::_scaled_dot_product_attention_math(query_, key, value, attn_mask_, dropout_p, need_attn_weights, is_causal);
-        }
+    const Tensor& query_,
+    const Tensor& key,
+    const Tensor& value,
+    const c10::optional<Tensor>& attn_mask_,
+    double dropout_p,
+    bool need_attn_weights,
+    bool is_causal) {
+  return at::_scaled_dot_product_attention_math(
+      query_,
+      key,
+      value,
+      attn_mask_,
+      dropout_p,
+      need_attn_weights,
+      is_causal);
+}
 
 std::tuple<Tensor, Tensor> _scaled_dot_product_attention_math(
         const Tensor& query_, const Tensor& key, const Tensor& value,
         const c10::optional<Tensor>& attn_mask_, double dropout_p, bool need_attn_weights, bool is_causal) {
+     // We are not training and we are falling back to math case.
+  // Inputs are required to be contiguous if nested
+    Tensor query_contiguous = query_;
+    Tensor key_contiguous = key;
+    Tensor value_contiguous = value;
+    if (query_.is_nested()) {
```

Comment:
We could also error out here

Reply:
I think that's a good idea for now if there's no risk we'll take this path from attention.cpp